src/main/packages/common/uml-component/uml-component-component.tsx

```diff
@@ -28,12 +28,12 @@ export const UMLComponentComponent: FunctionComponent<Props> = ({ element, child
       />
     </g>
     <Text fill={element.textColor}>
-      {element.stereotype && (
+      {element.stereotype && element.displayStereotype && (
         <tspan x="50%" dy={-8} textAnchor="middle" fontSize="85%">
           {`«${element.stereotype}»`}
         </tspan>
       )}
-      <tspan x="50%" dy={element.stereotype ? 18 : 10} textAnchor="middle">
+      <tspan x="50%" dy={element.stereotype && element.displayStereotype ? 18 : 10} textAnchor="middle">
```

Comment:
Whenever the stereotype is invisible, the vertical name alignment is off-center due to the `dy=10` offset. Same goes for the other elements using a stereotype.
Using a 0-offset looks much better here, see this comparison:
![Screenshot 2024-08-01 231650](https://github.com/user-attachments/assets/ea0735ea-4004-4526-ba46-a0bbc8473847)
![Screenshot 2024-08-01 231708](https://github.com/user-attachments/assets/17651041-e782-4d10-ab42-11d0566048be)